### DISCUSSION TITLE
No Cridential Received probably credential status for ....

### DISCUSSION THREAD

**Discussion by streetwalker-unity** [lupidan/apple-signin-unity#107] [Q&A]
(I'm running the iOS build the plugin's Sample Scene in Unity without any changes or additions):

After logging in to my iOS test app on my XCode build device and then closing the app on the phone, I'm getting this message in my app on subsequent runs when I launch the app again:

NO CREDENTIAL RECEIVED
Probably credential status for 08230rks ... asejr0u23r was Authorized.

Is this just a message telling me quick login worked, or is this an error?   If it is and error, how to fix it?

**Comment by lupidan** [comment]
Hi, that's expected.

When you first used Login with Apple ID, you get an Apple User ID plus the name/email/token etc... (those are the credentials)
After that, the next time you launch the app, you just need to use that Apple User ID (you should have saved it somewhere in the device), and [call](https://github.com/lupidan/apple-signin-unity#checking-credential-status) `GetCredentialState` with it.

If the result is Authorized, you know that user is still OK to access your app.
If the result is Revoked or Not Found, that Apple User ID is no longer valid, and you need to kick the user out, so they can login again.

Please familiarize yourself with the overall flow of how "Sign In With Apple" works:
https://github.com/lupidan/apple-signin-unity#implement-sign-in-with-apple

**Reply by streetwalker-unity** [reply]
Hey thanks - yes I suspected it meant it was working, it was just the nature of the message that was a bit confusing.

Yes, I have looked at that flowchart prior to posting my question.  Sorry, I'm a bit dense.   I see in the code there is a test in SetupAppleData that  if (receivedCredential == null)  is true, the message I posted is generated.   

Where in the flow chart does that test get tripped?

Thanks for your further insights!

**Comment by sudo-raza** [comment]
Hello, how do you handle the Apple UID if the person deletes the app and reinstalls it later? Do we need to store the UID on a server?

**Reply by lupidan** [reply]
The quick login should trigger another login attempt (no permissions asked) if the user has used the feature in the past successfully. Otherwise, Quick Login fails.

If a person did use Sign In With Apple to make a user on your app, deletes the app, and reinstalls it, then the Quick Login should detect it, let the user log in again, and give you the Apple UID back along with identity tokens.

Please check the flow chart:
https://github.com/lupidan/apple-signin-unity#implement-sign-in-with-apple

So to answer your question, I believe if you want to have a user sync on your server, you should validate the received identity token and use the Apple UID to identify which user is it.
